src/xenia/ui/window_gtk.cc

```diff
@@ -57,30 +52,80 @@ void gtk_event_handler_(GtkWidget* widget, GdkEvent* event, gpointer data) {
       window->HandleKeyboard(&(event->key));
       break;
     case GDK_SCROLL:
-    case GDK_BUTTON_PRESS:
     case GDK_MOTION_NOTIFY:
+    case GDK_BUTTON_PRESS:
+    case GDK_BUTTON_RELEASE:
       window->HandleMouse(&(event->any));
       break;
     case GDK_FOCUS_CHANGE:
       window->HandleWindowFocus(&(event->focus_change));
       break;
     case GDK_CONFIGURE:
-      window->HandleWindowResize(&(event->configure));
+      // Only handle the event for the drawing area so we don't save
+      // a width and height that includes the menu bar on the full window
+      if (event->configure.window ==
+          gtk_widget_get_window(window->drawing_area_)) {
+        window->HandleWindowResize(&(event->configure));
+      }
+      break;
     default:
       // Do nothing
-      return;
+      break;
   }
+  // Propagate the event to other handlers
+  return GDK_EVENT_PROPAGATE;
+}
+
+gboolean draw_callback(GtkWidget* widget, GdkFrameClock* frame_clock,
+                       gpointer data) {
+  GTKWindow* window = reinterpret_cast<GTKWindow*>(data);
+  window->HandleWindowPaint();
+  return G_SOURCE_CONTINUE;
+}
+
+gboolean close_callback(GtkWidget* widget, gpointer data) {
+  GTKWindow* window = reinterpret_cast<GTKWindow*>(data);
+  window->Close();
+  return G_SOURCE_CONTINUE;
 }
 
 void GTKWindow::Create() {
+  // GTK optionally allows passing argv and argc here for parsing gtk specific
+  // options. We won't bother
   window_ = gtk_window_new(GTK_WINDOW_TOPLEVEL);
-  gtk_window_set_title(GTK_WINDOW(window_), (gchar*)title_.c_str());
-  gtk_window_set_default_size(GTK_WINDOW(window_), 1280, 720);
-  gtk_widget_show_all(window_);
-  g_signal_connect(G_OBJECT(window_), "destroy", G_CALLBACK(gtk_main_quit),
-                   NULL);
-  g_signal_connect(G_OBJECT(window_), "event", G_CALLBACK(gtk_event_handler_),
+  gtk_window_set_resizable(GTK_WINDOW(window_), true);
+  gtk_window_set_title(GTK_WINDOW(window_), title_.c_str());
+  gtk_window_set_default_size(GTK_WINDOW(window_), width_, height_);
+  // Drawing area is where we will attach our vulkan/gl context
+  drawing_area_ = gtk_drawing_area_new();
+  // tick callback is for the refresh rate of the window
+  gtk_widget_add_tick_callback(drawing_area_, draw_callback,
+                               reinterpret_cast<gpointer>(this), nullptr);
+  // Attach our event handler to both the main window (for keystrokes) and the
+  // drawing area (for mouse input, resize event, etc)
+  g_signal_connect(G_OBJECT(drawing_area_), "event",
+                   G_CALLBACK(gtk_event_handler),
                    reinterpret_cast<gpointer>(this));
+
+  GdkDisplay* gdk_display = gtk_widget_get_display(window_);
+  assert(GDK_IS_X11_DISPLAY(gdk_display));
+  connection_ = XGetXCBConnection(gdk_x11_display_get_xdisplay(gdk_display));
```

Comment:
Is that the only XCB/Xlib-specific part of our window handling? It seems that we can use Xlib directly instead of Xlib-XCB if we only need the display and the window.

Reply:
Perhaps     
```c++
connection_ = XGetXCBConnection(gdk_display);
```

Reply:
Yes, this seems like the only place, so it looks like we can use VK_KHR_xlib_surface without any XCB dependencies.